.circleci/config.base.yml

```diff
@@ -31,6 +31,16 @@ clean_e2e_resources: &clean_e2e_resources
     yarn clean-e2e-resources job ${CIRCLE_BUILD_NUM}
   working_directory: ~/repo
 
+scan_e2e_test_artifacts: &scan_e2e_test_artifacts
+  name: Scan And Cleanup E2E Test Artifacts
+  command: |
+    if ! yarn ts-node .circleci/scan_artifacts.ts; then
+      echo "Cleaning the repository"
+      git clean -fdx
+      exit 1
+    fi
+  when: always
```

Comment:
I'm assuming this allows this step to run even if a previous step failed?

Reply:
nvm, I can read: https://circleci.com/docs/2.0/configuration-reference/#the-when-attribute

Reply:
That is correct. Here is an example where the test failed and the artifact scanning still happened.
![Screen Shot 2021-06-23 at 3 20 19 PM](https://user-images.githubusercontent.com/511386/123175368-8a630600-d436-11eb-8660-e1bd7dbf9e2a.png)

PS: Artifact scan failure is done intentionally to test it